src/sentry/utils/auth.py

```diff
@@ -448,3 +448,7 @@ def construct_link_with_query(path: str, query_params: dict[str, str]) -> str:
     if query_string:
         redirect_uri += f"?{query_string}"
     return redirect_uri
+
+
+class AuthenticatedHttpRequest(HttpRequest):
```

Comment:
might be worth a link to https://github.com/typeddjango/django-stubs?tab=readme-ov-file#how-can-i-create-a-httprequest-thats-guaranteed-to-have-an-authenticated-user